### PR TITLE
Add Pester tests for Windows Features

### PR DESCRIPTION
**Pull request by vsafonkin**
# Description
Added Pester tests:
- Windows features
- Disk space

#### Related issue:
[#805](https://github.com/actions/virtual-environments-internal/issues/805)

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/win/scripts/Tests/Tools.Tests.ps1 [hunk 1]
```diff
@@ -14,4 +14,30 @@ Describe "Kind" {
     It "Kind" {
         "kind version" | Should -ReturnZeroExitCode
     }
+}
+
+Describe "WindowsNETFeatures" {
```
Comment (maxim-lobanov) [inline images/win/scripts/Tests/Tools.Tests.ps1:19]: May be put those functions to the separate `Windows.Tests.ps1`. I don't think it is related to Tools

Comment (vsafonkin) [reply images/win/scripts/Tests/Tools.Tests.ps1]: Yes, move these test to file `WindowsFeatures.Tests.ps1`

File: images/win/scripts/Tests/WindowsFeatures.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    if (Test-isWin16) {
+        $testCases += @{ FeatureName = "BITS" }
+        $testCases += @{ FeatureName = "DSC-Service" }
+    }
+
+    It "Windows NET Feature <FeatureName> is installed" -TestCases $testCases {
+        (Get-WindowsFeature -Name $FeatureName).InstallState | Should -Be "Installed"
+    }
+
+
```
Comment (maxim-lobanov) [inline images/win/scripts/Tests/WindowsFeatures.Tests.ps1:18]: spaces

File: images/win/scripts/Tests/WindowsFeatures.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,25 @@
+Describe "WindowsNETFeatures" {
```
Comment (al-cheb) [inline images/win/scripts/Tests/WindowsFeatures.Tests.ps1:1]: Rename to more obvious  name
WindowsNETFeatures -> WindowsFeatures

File: images/win/scripts/Tests/WindowsFeatures.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        @{ FeatureName = "NET-Framework-45-Features" }
+        @{ FeatureName = "FS-iSCSITarget-Server" }
+    )
+
+    if (Test-isWin16) {
+        $testCases += @{ FeatureName = "BITS" }
+        $testCases += @{ FeatureName = "DSC-Service" }
+    }
+
+    It "Windows NET Feature <FeatureName> is installed" -TestCases $testCases {
```
Comment (al-cheb) [inline images/win/scripts/Tests/WindowsFeatures.Tests.ps1:14]: Remove NET

File: images/win/scripts/Tests/WindowsFeatures.Tests.ps1 (hunk 3 above)
Comment (maxim-lobanov) [inline images/win/scripts/Tests/WindowsFeatures.Tests.ps1:1]: Recently, feature `Microsoft-Windows-Subsystem-Linux` was enabled: https://github.com/actions/virtual-environments/pull/1081/files

Could you please add it here?

Comment (vsafonkin) [reply images/win/scripts/Tests/WindowsFeatures.Tests.ps1]: Fixed

File: images/win/scripts/Tests/WindowsFeatures.Tests.ps1 (hunk 4 above)
Comment (vsafonkin) [reply images/win/scripts/Tests/WindowsFeatures.Tests.ps1]: Fixed

File: images/win/scripts/Tests/WindowsFeatures.Tests.ps1 (hunk 3 above)
Comment (vsafonkin) [reply images/win/scripts/Tests/WindowsFeatures.Tests.ps1]: Added

File: images/win/scripts/Tests/WindowsFeatures.Tests.ps1 [hunk 5]
```diff
@@ -11,15 +11,25 @@ Describe "WindowsNETFeatures" {
...
 }
 
 Describe "ContainersFeature" {
     It "Windows containers feature is installed" {
         (Get-WindowsFeature -Name "Containers").InstallState | Should -Be "Installed"
     }
+}
+
+if (Test-isWin19) {
+    Describe "WSL" {
```
Comment (maxim-lobanov) [inline images/win/scripts/Tests/WindowsFeatures.Tests.ps1:26]: I think it is simple to just add WSL to the list above and avoid code duplication.
```
$testCases = @(
        @{ FeatureName = "NET-Framework-Features" }
        @{ FeatureName = "NET-Framework-45-Features" }
        @{ FeatureName = "FS-iSCSITarget-Server" }
    )

    if (Test-isWin16) {
        $testCases += @{ FeatureName = "BITS" }
        $testCases += @{ FeatureName = "DSC-Service" }
    }
    if (Test-isWin19) {
        $testCases += @{ FeatureName ="Microsoft-Windows-Subsystem-Linux" }
    }

Comment (vsafonkin) [reply images/win/scripts/Tests/WindowsFeatures.Tests.ps1]: done

**Pull request merged**